corehq/util/es/interface.py

```diff
@@ -1,195 +1,99 @@
+        return self.manager.index_put_mapping(index, doc_type, mapping)
 
     def _verify_is_alias(self, alias):
         """Verify that an alias is one of a registered index"""
         from corehq.apps.es.registry import verify_alias
         verify_alias(alias)
 
-    def update_index_settings(self, index, settings_dict):
-        assert set(settings_dict.keys()) == {'index'}, settings_dict.keys()
-        return self.es.indices.put_settings(settings_dict, index=index)
+    def update_index_settings_reindex(self, index):
+        self.manager.index_configure_for_reindex(index)
 
-    def _get_source(self, index_alias, doc_type, doc_id, source_includes=None):
-        kwargs = {"_source_include": source_includes} if source_includes else {}
-        return self.es.get_source(index_alias, doc_type, doc_id, **kwargs)
+    def update_index_settings_standard(self, index):
+        self.manager.index_configure_for_standard_ops(index)
 
     def doc_exists(self, index_alias, doc_id, doc_type):
-        return self.es.exists(index_alias, doc_type, doc_id)
-
-    def _mget(self, index_alias, body, doc_type):
-        return self.es.mget(
-            index=index_alias, doc_type=doc_type, body=body, _source=True)
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        return doc_adapter.exists(doc_id)
 
     def get_doc(self, index_alias, doc_type, doc_id, source_includes=None):
         self._verify_is_alias(index_alias)
-        doc = self._get_source(index_alias, doc_type, doc_id, source_includes=source_includes)
-        doc['_id'] = doc_id
-        return doc
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        if source_includes is None:
+            source_includes = []
+        return doc_adapter.fetch(doc_id, source_includes=source_includes)
 
     def get_bulk_docs(self, index_alias, doc_type, doc_ids):
-        from corehq.elastic import ESError
         self._verify_is_alias(index_alias)
-        docs = []
-        results = self._mget(index_alias=index_alias, doc_type=doc_type, body={'ids': doc_ids})
-        for doc_result in results['docs']:
-            if 'error' in doc_result:
-                raise ESError(doc_result['error'].get('reason', 'error doing bulk get'))
-            if doc_result['found']:
-                self._fix_hit(doc_result)
-                docs.append(doc_result['_source'])
-        return docs
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        return doc_adapter.fetch_many(doc_ids)
 
     def index_doc(self, index_alias, doc_type, doc_id, doc, params=None):
         self._verify_is_alias(index_alias)
-        self.es.index(index_alias, doc_type, body=self._without_id_field(doc), id=doc_id,
-                      params=params or {})
+        doc_adapter = self._get_doc_adapter(index_alias, doc_type)
+        if doc.get("_id", object()) != doc_id:
+            doc["_id"] = doc_id
```

Comment:
This is a confusing check. If `_id` is in the doc and not equal to `doc_id` then I think it would be better to raise and exception instead of just overwriting the value in the doc.

Reply:
This is functionally identical to what the previous implementation did. Removed logic was:
```python
self.es.index(index_alias, doc_type, body=self._without_id_field(doc), id=doc_id,
                      params=params or {})
```
Where
- `body=self._without_id_field(doc)` == _"strip `_id` from document"_
- `id=doc_id` == _"explicitly use `doc_id`"_

While fixing the possibility of that being a bug is outside the scope of this commit, you make a good point that this is concerning logic. I'll add a `TODO` here to make sure it gets revisited.